daffodil-propgen/src/main/resources/org/apache/daffodil/xsd/DFDL_part1_simpletypes.xsd

```diff
@@ -794,6 +794,17 @@
            </xsd:documentation>
          </xsd:annotation>
        </xsd:enumeration>
+       <xsd:enumeration value="midBitsToEnd">
+         <xsd:annotation>
+           <xsd:documentation><![CDATA[
+             Move the bits with indecis [x, x+y] to
+             the end of the layer block.
+
+             x,y are provided as a space delimated list in layerTransformArgs
```

Comment:
Positional arguments seem simpler (particularly if these arguments are going to be computed instead of hard coded). And we don't really have named arguements anywhere else in DFDL.

Reply:
DFDL doesn't really have positional arguments either. Every property has it's own DFDL type. Maybe the more DFDL-y way is to have a unique property for for each argument, e.g. dfdl:mbitsToEndFirstBitIndex, dfdl:mbitsToEndNumBits, but that's probably going to get messy, and would make pluggable transforms more difficult. I agree that positional arguments are definitely simpler (though, from an implementation perspective not significantly so), but they aren't as extensible. Adding new arguments later or deprecating old ones becomes diffiult. Optional args are impossible. And it's not clear from just looking at the layerTransformRags value what the args actually mean. So there's just some limitations.

But, maybe we don't need that flexablity. I guess this is fine for now, and if we realize it's too limiting or confusing in the real world, we can always change how things work. Perhaps we should consider extensions as a sort of "beta" featue that are subject to change, and so we don't have to worry quite so much about backwards compatability.

Reply:
DFDL does have DPath, which is where the only other function-ish feature I can think of is present in DFDL. In fact, the transformArgs itself is in DPath, and I imagine we will be adding more ways for DPath to influence layer transforms as time goes on.